Bottom: d54cf6574a24dbe12ec77b8d8be411e6ce0c163c
Top:    0eabf4b37d1fca5b44bb3537c11af535e3f57d32
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-26 20:05:29 +0100

Refresh of hid-gtk-fix-pinout-preview-def

---

diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index 46806da..88576a1 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -51,24 +51,19 @@
 
 RCSID ("$Id$");
 
-
+/* Just define a sensible scale, lets say (for example), 100 pixel per 150 mil */
+#define SENSIBLE_VIEW_SCALE  (100. / MIL_TO_COORD (150.))
 static void
-pinout_zoom_fit (GhidPinoutPreview * pinout, gint zoom)
+pinout_set_view (GhidPinoutPreview * pinout)
 {
-  pinout->zoom = zoom;
-
-  /* Should be a function I can call for this:
-   */
-  pinout->scale = 1.0 / (100.0 * exp (pinout->zoom * LN_2_OVER_2));
+  float scale = SENSIBLE_VIEW_SCALE;
 
   pinout->x_max = pinout->element.BoundingBox.X2 + Settings.PinoutOffsetX;
   pinout->y_max = pinout->element.BoundingBox.Y2 + Settings.PinoutOffsetY;
-  pinout->w_pixels = (gint) (pinout->scale *
-			     (pinout->element.BoundingBox.X2 -
-			      pinout->element.BoundingBox.X1));
-  pinout->h_pixels = (gint) (pinout->scale *
-			     (pinout->element.BoundingBox.Y2 -
-			      pinout->element.BoundingBox.Y1));
+  pinout->w_pixels = scale * (pinout->element.BoundingBox.X2 -
+                              pinout->element.BoundingBox.X1);
+  pinout->h_pixels = scale * (pinout->element.BoundingBox.Y2 -
+                              pinout->element.BoundingBox.Y1);
 }
 
 
@@ -109,7 +104,7 @@ pinout_set_data (GhidPinoutPreview * pinout, ElementType * element)
 		       Settings.PinoutOffsetY -
 		       pinout->element.BoundingBox.Y1);
 
-  pinout_zoom_fit (pinout, 3);
+  pinout_set_view (pinout);
 
   ELEMENTLINE_LOOP (&pinout->element);
   {
diff --git a/src/hid/gtk/gui-pinout-preview.h b/src/hid/gtk/gui-pinout-preview.h
index 978bb25..77fdd1b 100644
--- a/src/hid/gtk/gui-pinout-preview.h
+++ b/src/hid/gtk/gui-pinout-preview.h
@@ -52,8 +52,6 @@ struct _GhidPinoutPreview
   GtkDrawingArea parent_instance;
 
   ElementType element;		/* element data to display */
-  gfloat zoom;			/* zoom factor of window */
-  gfloat scale;			/* scale factor of zoom */
   gint x_max, y_max;
   gint w_pixels, h_pixels;	/* natural size of element preview */
 };
